index/index_monitor.go

```diff
@@ -0,0 +1,60 @@
+package index
+
+// helpers to atomically switch from using one index to another
+
+import (
+	"sync"
+)
+
+type IndexReference struct {
+	rwMutex sync.RWMutex
+	index *Index
+}
+
+// Atomically acquire a reference to an index, and increment the refcount
```

Comment:
These should be of the form `Get returns a reference to the index...`. `golint` will helpfully complain about that, if you run it (although I just realized there are some other golint errors I should fix...)